File: wrappers/dotnet/indy-sdk-dotnet/WalletApi/WalletItemNotFoundException.cs
namespace Hyperledger.Indy.WalletApi
{
    /// <summary>
    /// Exception thrown when value with the specified key doesn't exists in the wallet from which it was requested.
    /// </summary>
    /// <seealso cref="Hyperledger.Indy.IndyException" />
    public class WalletItemNotFoundException : IndyException
    {
        private const string message =
            "No value with the specified key exists in the wallet from which it was requested.";

        internal WalletItemNotFoundException() : base(message, (int)ErrorCode.WalletItemNotFoundError)
        {
        }
    }
}

File: wrappers/java/src/main/java/org/hyperledger/indy/sdk/wallet/WalletItemNotFoundException.java
package org.hyperledger.indy.sdk.wallet;

import org.hyperledger.indy.sdk.ErrorCode;
import org.hyperledger.indy.sdk.IndyException;

/**
 * Exception thrown when requesting a value from a wallet that does not contain the specified key.
 */
public class WalletItemNotFoundException extends IndyException
{
	private static final long serialVersionUID = 667964860056778208L;
	private final static String message = "No value with the specified key exists in the wallet from which it was requested.";

	/**
	 * Initializes a new WalletItemNotFoundException.
	 */
	public WalletItemNotFoundException() {
		super(message, ErrorCode.WalletItemNotFound.value());
	}
}
